packages/flutter/lib/src/widgets/feedback.dart

```diff
@@ -167,6 +168,4 @@ abstract final class Feedback {
       callback();
     };
   }
-
-  static TargetPlatform _platform(BuildContext context) => Theme.of(context).platform;
```

Comment:
This makes me wonder what we think the difference should be between `defaultTargetPlatform` and `ThemeData.platform`. Notice that this change could be breaking if any users are changing their theme platform but not their defaultTargetPlatform. This is probably another sign that we need some kind of theming at the `widgets` library level.

defaultTargetPlatform can only be changed with debugDefaultTargetPlatformOverride, which doesn't work in production builds. So by using defaultTargetPlatform we're saying that users can't change the feature, it must match the real platform. By using ThemeData.platform we're saying that it's valid for users to prefer the value of some other platform and use that in production instead of the platform default. Is that how we draw the line?

Reply:
Yeah, we chatted about this a bit yesterday.

I also thought it might be breaking to not defer to the theme, but then I realized - oh wait. Feedback is device-specific feedback actions like haptic buzzes, so it does not really apply to whatever the ThemeData says the platform is.

For example, if I am on a desktop machine, and Theme.of(context).platform has been configured to Android, the desktop won't actually produce haptic feedback. Currently Feedback only has implemented Android feedback options as well, so for these reasons it seemed like a safe change, and one that made the feature a bit more logical.. or at least less prone to Feedback that does not make sense for the actual device that is being used.

WDYT @justinmc?

> This is probably another sign that we need some kind of theming at the widgets library level.

I agree with this, I don't know if it applies here, but I am excited we are exploring this more and more. :)

Reply:
Proposal: Use ThemeData.platform if you can, unless it's something that affects platform calls, in which case use defaultTargetPlatform. So deciding which context menu to show should be a theme thing, but deciding to call the platform to show the native iOS context menu should use defaultTargetPlatform.

Currently it's not possible to use Theme in the widgets library, so I would also propose that long-term we should have a unified theme in the widgets library that Material and Cupertino derive their themes from.

In the case of this PR, following the proposal I would say that the question of which haptic vibration to use should be controlled by the theme. Since that's not yet possible, use defaultTargetPlatform in the meantime.

<hr />

So according to the proposal, this PR is correct, with the caveat that we'd like to migrate this to use some kind of widget theme in the future. That's another reason to keep the `BuildContext` parameters, as you have done. In the future we'll use that to get the theme.

Reply:
@Piinks Sorry, my previous comment was written without seeing your reply, I went to lunch and didn't refresh the page!

Reply:
Yeah actually this PR would affect platform channel calls, because [HapticFeedback directly calls platform channel methods](https://github.com/flutter/flutter/blob/8317bc2b16497217934ba7ab538b602d695d811f/packages/flutter/lib/src/services/haptic_feedback.dart#L20-L22) to cause vibration. But it still feels to me like setting the Theme to iOS while on Android in order to change the vibration is a valid thing to do. Maybe I need to rephrase my proposal...